nautobot/core/models/utils.py

```diff
@@ -164,20 +165,31 @@ def find_models_with_matching_fields(app_models, field_names, field_attributes=N
         (dict): A dictionary where the keys are app labels and the values are sets of model names.
     """
     registry_items = {}
+    field_names = field_names or []
     field_attributes = field_attributes or {}
     additional_constraints = additional_constraints or {}
     for model_class in app_models:
         app_label, model_name = model_class._meta.label_lower.split(".")
-        for field_name in field_names:
-            try:
-                field = model_class._meta.get_field(field_name)
-                if all((getattr(field, item, None) == value for item, value in field_attributes.items())) and all(
-                    getattr(model_class, additional_field, None) == additional_value
-                    for additional_field, additional_value in additional_constraints.items()
-                ):
-                    registry_items.setdefault(app_label, set()).add(model_name)
-            except FieldDoesNotExist:
-                pass
+        if field_names:
+            for field_name in field_names:
+                try:
+                    field = model_class._meta.get_field(field_name)
+                    if all((getattr(field, item, None) == value for item, value in field_attributes.items())) and all(
+                        getattr(model_class, additional_field, None) == additional_value
+                        for additional_field, additional_value in additional_constraints.items()
+                    ):
+                        registry_items.setdefault(app_label, set()).add(model_name)
+                except FieldDoesNotExist:
+                    pass
+        else:
+            # SavedView does not add any specific fields to the model
+            # so we need to look at the additional constraints only.
+            if all(
+                getattr(model_class, additional_field, None) == additional_value
+                for additional_field, additional_value in additional_constraints.items()
+            ):
+                registry_items.setdefault(app_label, set()).add(model_name)
```

Comment:
Feel like it might be possible to refactor this a bit further... for one, I'm realizing that the implementation doesn't match the docstring, in that the docstring says `field_names` is "fields that must be present" (implying "all") but the actual implementation adds models that have *any* of the fields in `field_names`.

Perhaps something like:

```suggestion
        valid_model = True
        for field_name in field_names:
            try:
                field = model_class._meta.get_field(field_name)
                if not all((getattr(field, item, None) == value for item, value in field_attributes.items())):
                    valid_model = False
                    break
            except FieldDoesNotExist:
                valid_model = False
                break
        if valid_model:
            if not all(
                getattr(model_class, additional_field, None) == additional_value
                for additional_field, additional_value in additional_constraints.items()
            ):
                valid_model = False
        if valid_model:
            registry_items.setdefault(app_label, set()).add(model_name)
```